### PR TITLE
fix: skip automatic id tiebreaker for unmapped or non-identity models (#305)

### LINKED ISSUE #305: DefaultSortQueryBuilder Appends id field even if model doesn't have an id field e.g., logstash index.
// Logstash indices are created/mapped entirely by the external Logstash pipeline, which
  // never writes an `id` field. Foundatio.Repositories >= 7.18.0 unconditionally appends an
  // `id` tie-breaker sort (DefaultSortQueryBuilder) to every query; against an index with no
  // `id` mapping at all, Elasticsearch rejects the sort with `all shards failed`

### PR DESCRIPTION
**Pull request by niemyjski**
Fixes #305. Queries against externally managed indexes could receive an automatic `id` sort even when the model had no identity or the server mapping did not support that sort. The PR makes ID sorting explicit and keeps Live and point-in-time cursor traversal consistent across forward, backward, and failure paths.

## Changes

- Share ID-tiebreaker validation through `IdTiebreakerField.TryEnsure<T>()`. Models without `IIdentity` skip the automatic ID sort. `IIndex.HasSortableIdField` defaults to `true` for existing consumers; externally managed indexes can opt out. Explicit sorts remain supported.
- Let daily/monthly adapters override `MappingIndexPattern` and `GetIndexDate()` to select the latest valid external partition. Ignore malformed dated matches and log when no mapping matches. Include a complete constructor, mapping pattern, date parser, and ID-sort opt-out example.
- Validate cursor direction/cardinality and incompatible paging modes after `BeforeQuery`. Live paging requires a usable sort; PIT paging includes the reversible `_shard_doc` tiebreaker. Preserve existing cursor return annotations.
- Treat failed PIT searches, including HTTP 404, as `DocumentException` while retaining empty results for ordinary missing-index searches. Cleanup uses the latest PIT ID and preserves the original exception. Caller-owned PITs are never closed automatically. Continuations retain their original session identity across ID rotation and validate it after `BeforeQuery`; clearing or replacing a session cannot switch to live paging or attach old results to a newly opened PIT. Retained sessions retry with the latest ID and correct page number. Active requests retain cleanup responsibility if event handlers clear an owned session; an explicit `PointInTimeId(...)` handoff preserves caller ownership.
- Request complete results for Live and PIT cursors with `allow_partial_search_results=false`. Reject HTTP 200 timeout/shard-failure responses before returning hits, invoking `AfterQuery`, or advancing the cursor; include failure diagnostics.
- Bypass cursor-result caching, including Live cursors in `FindOneAsync` and `CountAsync`. Scalar `BeforeQuery` handlers run before cache decisions, including cache hits; ordinary queries still cache normally. Scalar PIT restrictions remain enforced.
- Reverse request-owned copies of every supported sort variant, preserving all settings including `FieldSort.Format`. The public `ReverseOrder` helper retains its in-place behavior.
- Synchronize consumer documentation, XML documentation, and repository skill guidance.

## Validation

- Final commit: `f7b26a8a0aab293755044b4c475c7d50e81da2dd`.
- Solution builds for .NET 8 and .NET 10: zero warnings and errors.
- Core suite: 152 passed, no failures or skips.
- Complete Elasticsearch suite on **each** isolated Elasticsearch 8.19.15 and 9.5.0: **891 passed**, no failures or skips.
- Added 41 further regression cases for session replacement, event hooks, explicit ownership transfer, incomplete responses in both directions, and retry behavior. Nine initial probes failed on the previous commit; another test reproduced orphan cleanup after a successful event hook. Earlier expired-PIT, cache, sort-copy, and real open-context cleanup regressions remain green.
- Actual local NuGet consumer builds for both frameworks with nullable warnings treated as errors. The complete external-index documentation example, legacy default-interface implementation, and nullable cursor call sites compile; execution passes on .NET 8.0.30 and .NET 10.0.11.
- Changed-file formatting and `git diff --check` pass; public signatures and cursor return annotations remain unchanged.
- GitHub CI on the final commit: both build/test jobs, Actions/C# analysis, CodeQL, and CLA checks pass. [Push build](https://github.com/FoundatioFx/Foundatio.Repositories/actions/runs/34076870426), [PR build](https://github.com/FoundatioFx/Foundatio.Repositories/actions/runs/34076874635).

## Cross-PR release reconciliation

An isolated combination of this commit, Repositories #307 at `751fa85f39e55f8346d8e34c71f99e34767d48d1`, and locally packaged Parsers #253 at `9f42b215410ade5c174ebd5db6d569b7352ecd26` builds cleanly and passes **1,138 runnable tests on each Elasticsearch version**. The one skipped case is #307's explicitly gated persistent 7→8→9 upgrade-chain test. Combined NuGet consumers also build for both frameworks with warnings treated as errors and execute on .NET 8.0.30 and .NET 10.0.11.

`DailyIndex.GetLatestIndexMapping()` has a merge conflict with #307. The tested reconciliation keeps `MappingIndexPattern` authoritative, preserves deterministic dated selection and open/hidden discovery, excludes owned error partitions, and requires canonical ownership for generated compatibility wrappers. Its candidate predicate is:

```csharp
.Where(i => !i.Value.Aliases.HasExactHiddenAlias(ElasticReindexer.ErrorIndexOwnershipAlias)
    && (CompatibilityIndexName.GetCanonicalNameSpan(i.Key, Name).Length == i.Key.Length
        || MatchesCompatibilitySource(i.Key, i.Value.Aliases)))
```

Keep this resolution when combining the PRs. Regression fixtures cover unversioned external mappings, a natural name beginning with `reindexed`, and Live/PIT traversal through upgraded canonical aliases. This reconciliation was tested in isolation; neither GitHub PR was merged or rebased, and no packages were published.

### CODE REVIEW

File: src/Foundatio.Repositories.Elasticsearch/Repositories/ElasticReadOnlyRepositoryBase.cs [hunk 1]
```diff
@@ -548,14 +598,32 @@ public async Task RemoveQueryAsync(string queryId)
...
+            options.Values.Remove(SearchAfterQueryExtensions.PointInTimeContinuationKey);
+        }
+    }
 
-        options.PageNumber(!options.HasPageNumber() ? 2 : options.GetPage() + 1);
-        return await FindAsAsync<TResult>(query, options).AnyContext();
+    private static void ValidatePointInTimeContinuation(ICommandOptions options, PointInTimeState? continuation)
+    {
+        if (continuation is not null &&
+            (!options.ShouldUseSearchAfterPagingPointInTime() || !ReferenceEquals(continuation, options.GetPointInTimeState())))
```
[the remark this answers is not in the document]
Comment (niemyjski) [reply src/Foundatio.Repositories.Elasticsearch/Repositories/ElasticReadOnlyRepositoryBase.cs]: Fixed on the PR in `02f88d366236802cb14464e0d772459216dda797`, with the regression tests in `629b7062e90b8551d42d260c36b7a9f6ea01f884`. The branch now contains the complete audited series through `e786babd17585db1d3a1ffca4402c5d611972be5`.

Live search-after pages now retain a session identity, just as PIT pages do. Continuations validate the originating session before mutating cursor/page options and again after `BeforeQuery`, while retaining virtual `FindAsAsync` dispatch. Disabling, switching modes, and disabling/re-enabling the same mode invalidate old pages; resetting inside `AfterQuery` also invalidates the returned continuation. Reapplying the current mode preserves the identity, and a failed Live request can retry the same cursor/page.

Evidence: the pre-fix run reproduced 16 expected failures (`No exception was thrown`) with 84 passing controls/existing tests: https://github.com/FoundatioFx/Foundatio.Repositories/actions/runs/35751059940/job/106825006770. The 21 new Live cases cover external resets, replacement searches, both query hooks, reapplication, and retries. All 144 request-level cases pass on the final candidate. The complete suite on Elasticsearch 8.19.15 also passed on this exact commit: 152 core + 957 Elasticsearch tests, zero failures/skips, in https://github.com/FoundatioFx/Foundatio.Repositories/actions/runs/35765843225/job/106876357991. The Elasticsearch 9 and documentation jobs are still pending at this update; this reply does not claim those checks passed.
